Monal/Classes/MediaGallery.swift

```diff
@@ -0,0 +1,240 @@
+            guard let fileExtension = fileInfo["fileExtension"] as? String else {
+                DDLogInfo("Could not get file extension for file, not generating thumbnail...")
+                return nil
+            }
+
+            //create a symlink for our file having the proper file extension
+            //the "tmp." prefix will make sure this gets garbage collected by doStartupCleanup in MLFiletransfer
+            let symlinkPath = HelperTools.getContainerURL(forPathComponents:["documentCache", "tmp.player_symlink.\(moviePath.lastPathComponent).\(fileExtension)"])
+            do {
+                if FileManager.default.fileExists(atPath: symlinkPath.path) {
+                    try FileManager.default.removeItem(at: symlinkPath)
+                }
+                try FileManager.default.createSymbolicLink(at: symlinkPath, withDestinationURL: moviePath)
+            } catch {
+                DDLogError("Error creating symlink: \(error)")
+                return nil
+            }
+            defer {
+                if FileManager.default.fileExists(atPath: symlinkPath.path) {
+                    try? FileManager.default.removeItem(at: symlinkPath)
+                }
+            }
+
+            //now generate the preview using the symlink
+            let asset = AVURLAsset(url: symlinkPath)
+            let imageGenerator = AVAssetImageGenerator(asset: asset)
+            imageGenerator.appliesPreferredTrackTransform = true
+            let time = CMTime(seconds: seconds, preferredTimescale: 600)
+            do {
+                let (cgImage, _) = try await imageGenerator.image(at: time)
+                return UIImage(cgImage: cgImage)
+            } catch {
+                DDLogError("Error generating thumbnail: \(error)")
+                return nil
+            }
+        }
+
+        //generate a thumbnail using the modern ios 17 method to attach a mime type to an AVURLAsset
+        let asset = AVURLAsset(url: moviePath, options:[AVURLAssetOverrideMIMETypeKey: fileInfo["mimeType"] as! String])
+        let imageGenerator = AVAssetImageGenerator(asset: asset)
+        imageGenerator.appliesPreferredTrackTransform = true
+        let time = CMTime(seconds: seconds, preferredTimescale: 600)
+        do {
+            let (cgImage, _) = try await imageGenerator.image(at: time)
+            return UIImage(cgImage: cgImage)
+        } catch {
+            DDLogError("Error generating thumbnail: \(error)")
+            return nil
+        }
+    }
+}
+
+struct MediaItemView: View {
+    @ObservedObject var item: MediaItem
+
+    var body: some View {
+        Group {
+            if item.thumbnail == nil {
+                ProgressView()
+                    .progressViewStyle(CircularProgressViewStyle())
+                    .frame(width: 50, height: 50)
+            } else if let thumbnail = item.thumbnail {
+                Image(uiImage: thumbnail)
+                    .resizable()
+                    .scaledToFill()
```

Comment:
Your if statement is not clean and the else case is never reached at all, consider cleaning this up:
```suggestion
            if if let thumbnail = item.thumbnail {
                Image(uiImage: thumbnail)
                    .resizable()
                    .scaledToFill()
            } else {
                ProgressView()
                    .progressViewStyle(CircularProgressViewStyle())
                    .frame(width: 50, height: 50)
            }
```